storm-core/src/jvm/backtype/storm/task/TopologyContext.java

```diff
@@ -205,32 +217,65 @@ public Object getTaskData(String name) {
     public void setExecutorData(String name, Object data) {
         _executorData.put(name, data);
     }
-    
+
     public Object getExecutorData(String name) {
         return _executorData.get(name);
-    }    
-    
+    }
+
     public void addTaskHook(ITaskHook hook) {
         hook.prepare(_stormConf, this);
         _hooks.add(hook);
     }
-    
+
     public Collection<ITaskHook> getHooks() {
         return _hooks;
     }
+
+    public Object groupingToJSONableObject(Grouping grouping) {
+    	if (grouping.is_set_fields()) {
+    		return grouping.get_fields();
+    	} else {
+    		return grouping.getSetField().toString();
+    	}
```

Comment:
I'm not sure I really like the way this is overloaded.  I'm not really sure how this is going to be used in the future, but I would prefer to have a way to cleanly extend things in the future, and checking type feels less ideal to me.  I would rather see something like 
```
{
    'type': 'SHUFFLE',
    'fields': ['a','b']
}
```

for this type would be required and would be the set field, and the each grouping could provide more data if needed in the future.